src/System.Device.Gpio/System/Device/Gpio/BoardIdentification .cs

```diff
@@ -0,0 +1,276 @@
+
+            /// <summary>
+            /// Pi 3 Model B.
+            /// </summary>
+            RaspberryPiB3,
+
+            /// <summary>
+            /// Pi 3 Model B+.
+            /// </summary>
+            RaspberryPiB3Plus,
+
+            /// <summary>
+            /// Compute module 3.
+            /// </summary>
+            RaspberryPiComputeModule3,
+
+            /// <summary>
+            /// Pi 4 all versions.
+            /// </summary>
+            RaspberryPi4,
+        }
+        #region Fields
+
+        private readonly Dictionary<string, string> _settings;
+
+        private BoardIdentification(Dictionary<string, string> settings)
+        {
+            _settings = settings;
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Get board model from firmware revision
+        /// See http://www.raspberrypi-spy.co.uk/2012/09/checking-your-raspberry-pi-board-version/ for information.
+        /// </summary>
+        /// <returns></returns>
+        internal Model GetBoardModel()
+        {
+            var firmware = Firmware;
+            switch (firmware & 0xFFFF)
+            {
+                case 0x2:
+                case 0x3:
+                    return Model.RaspberryPiBRev1;
+
+                case 0x4:
+                case 0x5:
+                case 0x6:
+                case 0xd:
+                case 0xe:
+                case 0xf:
+                    return Model.RaspberryPiBRev2;
+
+                case 0x7:
+                case 0x8:
+                case 0x9:
+                    return Model.RaspberryPiA;
+
+                case 0x10:
+                    return Model.RaspberryPiBPlus;
+
+                case 0x11:
+                    return Model.RaspberryPiComputeModule;
+
+                case 0x12:
+                    return Model.RaspberryPiAPlus;
+
+                case 0x1040:
+                case 0x1041:
+                    return Model.RaspberryPiB2;
+
+                case 0x0092:
+                case 0x0093:
+                    return Model.RaspberryPiZero;
+
+                case 0x00C1:
+                    return Model.RaspberryPiZeroW;
+
+                case 0x2082:
+                    return Model.RaspberryPiB3;
+
+                case 0x20D3:
+                    return Model.RaspberryPiB3Plus;
+
+                case 0x20A0:
+                    return Model.RaspberryPiComputeModule3;
+
+                case 0x3111:
+                    return Model.RaspberryPi4;
+
+                default:
+                    return Model.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Gets the processor name.
+        /// </summary>
+        /// <value>
+        /// The name of the processor.
+        /// </value>
+        public string ProcessorName
+        {
+            get
+            {
+                string hardware;
+                return _settings.TryGetValue("Hardware", out hardware) ? hardware : null;
```

Comment:
Just saw in a different context: The value returned for "Hardware" in /proc/cpuinfo is meaningless. It is always "BCM2835", which is the CPU of the Raspberry Pi 1.

Reply:
Which context? Whic board?

Reply:
This may be related to your comment above, and the question probably is what we expect. I get "Hardware: BCM2835" when I run `cat /proc/cpuinfo` on my Pi4. But what I'd actually expect to see as processor type would be BCM2711, as described [here](https://www.raspberrypi.org/documentation/hardware/raspberrypi/README.md). But I don't know whether that information is available.

Reply:
I see. I have had a similar problem with CM3 light version. but not with the CM3.
To check it I need to buy a PI4. In the meanwhile may you check if the cpu result is like this?

`
processor : 0
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 1
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 2
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 3
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

Hardware : BCM2835
Revision : c03111
`

Keep in mind that the only useful information for using the corrcet driver is the last one:
`Revision : c03111`

Reply:
The result is 
```
processor	: 0
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 1
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 2
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 3
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

Hardware	: BCM2835
Revision	: c03111
Serial		: 10000000adb9246b
Model		: Raspberry Pi 4 Model B Rev 1.1
```

Perhaps also have a look at #766: There is/was an issue that `/proc/cpuinfo` may not work when the Pi was not running Raspbian (but some other flavor of Linux).

Reply:
> There is/was an issue that /proc/cpuinfo may not work when the Pi was not running Raspbian (but some other flavor of Linux).

What you suggest to push ahead with this PR and board recognition? I would like to complete the driver.

Reply:
I was only saying that the detection method could be improved as demonstrated there. But it's actually not directly related to your changes. 
However, there's still something wrong with your PR, all the builds are red. Check the logs there and/or run the build locally until everything passes.

Reply:
@pgrawehr May you help me in understanding Checks log? 
I symply forked the project and compile it succefully on my PC. But some checks are unintelligible. For example this one:
`artifacts\obj\System.Device.Gpio\Release\netstandard2.0\System.Device.Gpio.notsupported.cs(37,57): error CS0234: The type or namespace name 'IDisposable' does not exist in the namespace 'System.Device.Gpio.System' (are you missing an assembly reference?)`

I do not have a namespace `'System.Device.Gpio.System` otherwise I couldn't compile it.
Thank you

Reply:
That error message indeed is a bit weird. It has likely something to do with the way the build works. I'll check later.